dcos/cluster.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from dcos import config, constants, http, util
+from dcos.errors import DCOSException
+
+logger = util.get_logger(__name__)
+
+
+def create_cluster_config():
+    """Create a cluster specific config file + directory
+    from a global config file. This will move users from global config
+    structure (~/.dcos/dcos.toml) to the cluster specific one
+    (~/.dcos/clusters/CLUSTER_ID/dcos.toml) and set that cluster as
+    the "attached" cluster.
+
+    :rtype: None
+    """
+
+    dcos_url = config.get_config_val("core.dcos_url")
+
+    # if no cluster is set, do not move the cluster yet
```

Comment:
The dcos_url is not unset by this. If the user doesn't have a dcos_url or it's invalid we skip the move to "cluster" configs. No need for an error message because this is a structure move that the user shouldn't have to know about.